### PR TITLE
Problem: DecodeTxResponses don't ignore non-eth msgs

### PR DESCRIPTION
**Pull request by mmsqe**
this change should have been included in 199ccc232fd4cc5b0468b0cdec5e1ddaa340e116

Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/types/utils.go [hunk 1]
```diff
@@ -55,14 +55,17 @@ func DecodeTxResponses(in []byte) ([]*MsgEthereumTxResponse, error) {
...
 		return nil, err
 	}
-
-	responses := make([]*MsgEthereumTxResponse, len(txMsgData.MsgResponses))
-	for i, res := range txMsgData.MsgResponses {
+	responses := make([]*MsgEthereumTxResponse, 0)
+	for _, res := range txMsgData.MsgResponses {
 		var response MsgEthereumTxResponse
-		if err := proto.Unmarshal(res.Value, &response); err != nil {
+		if res.TypeUrl != "/"+proto.MessageName(&response) {
```
Comment (yihuang) [inline x/evm/types/utils.go:61]: do you mean a non-eth msg could be bundled together with a eth msg in a single tx?

File: x/evm/types/utils.go [hunk 2]
```diff
@@ -55,14 +55,17 @@ func DecodeTxResponses(in []byte) ([]*MsgEthereumTxResponse, error) {
 	if err := proto.Unmarshal(in, &txMsgData); err != nil {
 		return nil, err
 	}
-
-	responses := make([]*MsgEthereumTxResponse, len(txMsgData.MsgResponses))
-	for i, res := range txMsgData.MsgResponses {
+	responses := make([]*MsgEthereumTxResponse, 0)
```
Comment (yihuang) [inline x/evm/types/utils.go]:
```suggestion
	responses := make([]*MsgEthereumTxResponse, 0, len(txMsgData.MsgResponses))
```

we'll need the capacity most of the time

File: x/evm/types/utils.go (hunk 1 above)
Comment (mmsqe) [reply x/evm/types/utils.go:61]: [yea](https://github.com/crypto-org-chain/cronos/actions/runs/6259692939/job/16998335986?pr=1163), and we have diff ibc proto in host and controller, and wrong wireType = 0 for field Hash trigger this bug

Comment (yihuang) [reply x/evm/types/utils.go:61]: I see, the tx is not necessarily a eth tx, when get logs from a block.

**Review by yihuang — APPROVED** [review]

**Pull request merged**